src/metabase/lib/drill_thru/zoom_in_timeseries.cljc

```diff
@@ -41,20 +43,19 @@
    [metabase.lib.schema :as lib.schema]
    [metabase.lib.schema.common :as lib.schema.common]
    [metabase.lib.schema.drill-thru :as lib.schema.drill-thru]
-   [metabase.lib.schema.metadata :as lib.schema.metadata]
    [metabase.lib.schema.temporal-bucketing :as lib.schema.temporal-bucketing]
    [metabase.lib.temporal-bucket :as lib.temporal-bucket]
    [metabase.lib.util :as lib.util]
    [metabase.shared.util.i18n :as i18n]
    [metabase.util.malli :as mu]))
 
-;;; TODO -- we shouldn't include hour and minute for `:type/Date` columns.
-(def ^:private valid-current-units
-  [:year :quarter :month :week :day :hour :minute])
-
-(def ^:private unit->next-unit
-  (zipmap (drop-last valid-current-units)
-          (drop 1 valid-current-units)))
+(mu/defn- valid-current-units :- [:vector ::lib.schema.temporal-bucketing/unit.date-time.truncate]
+  [query :- ::lib.schema/query
+   stage :- :int
+   field :- :mbql.clause/field]
+  (into [] (reverse (eduction (map lib.temporal-bucket/raw-temporal-bucket)
+                              (filter #(contains? lib.schema.temporal-bucketing/datetime-truncation-units %))
+                              (lib.temporal-bucket/available-temporal-buckets query stage field)))))
```

Comment:
This is a lot harder to read than a direct `->>` or `(reverse (into [] (comp (map ...) (filter ...)) ...))`.

Also, note that sets can be used as functions, so there's no need for `contains?` here; just `(filter lib.schema.temporal-bucketing/datetime-truncation-units)`.

You could write this as
```clojure
(reverse (into [] (comp (map lib.temporal-bucket/raw-temporal-bucket)
                        (filter lib.schema.temporal-bucketing/datetime-truncation-units))
                  (lib.temporal-bucket/available-temporal-buckets query stage field)))
```

but I honestly think it'll be better just to use `->>` naively. This list is too short to worry about any overhead.

I'll also point out in passing that `(comp f g)` applies `g`, then `f`, ie. it's right-to-left. **But** transducers are functions that wrap an abstract pipeline with a new, pre-order transformation. So `(comp (map f) (filter g))` produces a transducer that maps `f` and then filters with `g`.

```clojure
(->> (lib.temporal-bucket/available-temporal-buckets query stage field)
     (map lib.temporal-bucket/raw-temporal-bucket)
     (filter lib.schema.temporal-bucketing/datetime-truncation-units)
     reverse)
```

Reply:
Awesome, thanks for the tip. As you can tell I'm still struggling with idiomatic Clojure, so these tips are appreciated. Will update the PR with these suggestions.

Reply:
Fixed